source/core/Variants.cpp

```diff
@@ -86,6 +77,18 @@ bool IsTargetTypeCompatibleWithInput(TypeRef inputType, TypeRef targetType, void
     return inputType && (!targetData || inputType->IsA(targetType, true));
 }
 
+void SetVariantToDataTypeError(TypeRef inputType, TypeRef targetType, void* targetData, TypeRef outputType, void* outputData, ErrorCluster* errPtr)
```

Comment:
Make _ParamImmediateDef(StaticTypeAndData, TargetType) as StaticType instead as we don't care about data in that anymore.

Reply:
Like I mentioned the other day, we do require access to the data to check whether the terminal is connected to anything. If not (if the data is null), we should treat the target type as a variant. We cannot handle this case if we make it just StaticType. Look at how targetData is used below.

Reply:
I thought we were going to (or already doing with your change) to emit a variant constant (from DFIR) if the "type" terminal is unwired in G. It is better to do that now, as that will become a required terminal in G at some point in the future. Then, we can make this terminal as a "required" input and not an "optional" input for this primitive in VIA. I think that will be a lot cleaner.

Reply:
I am doing that, but it seemed like a good idea to have an extra check to guard against hand-written via that doesn't adhere to the insert-typed-local-if-null requirement. 

Are you worried that we would accidentally use the data from the type terminal? It doesn't seem like a breach of contract to me, since we are wiring in a type+data instance.